build/docker-entrypoint.sh

```diff
@@ -1,19 +1,27 @@
 #!/usr/bin/env bash
 
-if [ $1 == "dst-server" ]; then
-	set -e
+# Exit immediately on non-zero return codes.
+set -e
 
-	# Configure the server
+# Run boot scripts before starting the server.
+if [ $1 = 'dst-server' ]; then
+
+	# Prepare the shard directory.
 	mkdir -p $CLUSTER_PATH/$SHARD_NAME
 	chown -R $STEAM_USER:$STEAM_USER $CLUSTER_PATH
 
-	if [ "$2" == "start" ]; then
+	# Create configuration files.
+	if [ "$2" = 'start' ]; then
 		$DSTA_HOME/boot/token.sh
 		$DSTA_HOME/boot/settings.sh
 		$DSTA_HOME/boot/lists.sh
 		$DSTA_HOME/boot/world.sh
 		$DSTA_HOME/boot/mods.sh
 	fi
+
+	# Run as user `steam` if the command is `dst-server`.
+	set -- gosu steam "$@"
```

Comment:
With this changes this will not work:

```bash
docker exec -it container bash
dst-server version
```

You are running dst-server with root user. I prefer to check the correct user inside the files and try to change it.